scipy/stats/_discrete_distns.py

```diff
@@ -887,6 +887,74 @@ def _stats(self, mu1, mu2):
 skellam = skellam_gen(a=-np.inf, name="skellam", longname='A Skellam')
 
 
+class yulesimon_gen(rv_discrete):
+    r"""A Yule-Simon discrete random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+
+    The probability mass function for the `yulesimon` is:
+
+    .. math::
+
+        f(k) =  \alpha B(k, \alpha+1)
+
+    for :math:`k=1,2,3,...`, where :math:`\alpha>0`.
+    Here :math:`\B(k, \alpha+1)` refers to the `scipy.special.beta` function.
+
+    For details see: https://en.wikipedia.org/wiki/Yule-Simon_distribution
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _rvs(self, alpha):
+        """Uses method on 6.3 on pg 593 of Devroye, non-uniform random
+           variate generation to sample a Yule-Simon.
+           Here our notation maps to Devroye's via :math:`\alpha=a-1`.
+        """
+        E1 = self._random_state.standard_exponential(self._size)
+        E2 = self._random_state.standard_exponential(self._size)
+        ans = ceil(-E1 / log1p(-exp(-E2 / alpha)))
+        return ans
+
+    def _pmf(self, x, alpha):
+        return alpha * special.beta(x, alpha + 1)
+
+    def _argcheck(self, alpha):
+        return (alpha > 0)
+
+    def _logpmf(self, x, alpha):
+        return log(alpha) + special.betaln(x, alpha + 1)
+
+    def _cdf(self, x, alpha):
+        return 1 - x * special.beta(x, alpha + 1)
+
+    def _sf(self, x, alpha):
+        return x * special.beta(x, alpha + 1)
+
+    def _logsf(self, x, alpha):
+        return log(x) + special.betaln(x, alpha + 1)
+
+    def _stats(self, alpha):
+        if alpha <= 1:
+            return np.inf, np.inf, np.inf, np.inf
+        elif alpha <= 2 and alpha > 1:
+            return alpha / (alpha - 1), np.inf, np.inf, np.inf
+        elif alpha <= 3 and alpha > 2:
+            return alpha / (alpha - 1), alpha**2 / ((alpha - 2) * (alpha - 1)**2), np.inf, np.inf
```

Comment:
[PEP 8](https://www.python.org/dev/peps/pep-0008/): Keep line length to 79 characters or fewer.

Reply:
@WarrenWeckesser, thanks for the feedback, I didn't mention in commit explicitly that the commit is WIP. I was addressing @chrisb83 comments on doc strings but haven't got to the _stats method comment in his review yet. Once I put in `np.where` and re-org code so each moment is stored and all are returned (like in `t_gen` in _continuous_distns.py LIC: 5142-5153 that should reduce the redundancy, enhance readibility, and enforce PEP8 line length.